src/main/java/com/google/sps/servlets/FindLocationServlet.java

```diff
@@ -19,7 +27,25 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     
-    //TODO - implement the rest of this function (@Kaleab)
+    String correlator = request.getParameter("correlator");
+
+    //TODO: find a more graceful way to handle this
+    if (correlator == null){
+      throw new IOException();
+    }
+    
+    Filter correlatorFilter =  new FilterPredicate("correlator", FilterOperator.EQUAL, correlator);
+    Query impressionQuery = new Query("Impressions").setFilter(correlatorFilter);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery filteredImpression = datastore.prepare(impressionQuery);
+
+    //only runs once
+    for (Entity impression : filteredImpression.asIterable()) {
+      impression.setProperty("clicksFindNearestLocation", true);
```

Comment:
use a const for clicksFindNearestLocation.  Can you also comment why you are adding this field to your data?

Reply:
Added it to the class javadoc